lib/reporters/postman/index.js

```diff
@@ -0,0 +1,76 @@
+const _ = require('lodash'),
+    print = require('../../print'),
+    util = require('./helpers/util'),
+    uploadUtil = require('./helpers/upload-run');
+
+/**
+ * Reporter to upload newman run data to Postman servers
+ *
+ * @param {Object} newman - The collection run object with event handling hooks to enable reporting.
+ * @param {Object} _reporterOptions -  A set of reporter specific options.
+ * @param {*} collectionRunOptions - A set of generic collection run options.
+ * @returns {*}
+ */
+function PostmanReporter (newman, _reporterOptions, collectionRunOptions) {
+    newman.on('beforeDone', (error, o) => {
+        if (error || !_.get(o, 'summary')) {
+            return;
+        }
+
+        // We get the collection id in the collectionRunOptions. But that collection id has the user id stripped off
+        // Hence, we try to parse the CLI args to extract the whole collection id from it.
+        const processArgs = process.argv,
```

Comment:
How will this work when Newman is used as a library?

Reply:
This will be available only via CLI.
The issue is we need the collection UID to publish runs (wrt a collection) and we can get it from CLI args but when used as a library we don't have a way to get it. (The reporter gets passed the collection object fetched via postman api which doesnt have the UID)
 
Will figure out a way to use it as part of library also in the second iteration.
(Our current target use-case is CLI invocations as part of CI/CD)

Reply:
@codenirvana have made the changes to make the reporter compatible with library also.
cc @anshuljain03 @shubhbhargav